### PR TITLE
ci: improve parallelism, add windows for e2e tests

### PR DESCRIPTION
**Pull request by johnpc**
#### Description of changes

This PR accomplishes two major goals that improve our CI process:

1. Windows coverage is added
2. Multi-account AWS Organization strategy is used to enable scalable parallelization of tasks

Many windows tests are skipped for now (excluded via split-e2e-tests script), since there are issues with the test harness compatibility that will need to be addressed one by one. Still, even minimal windows coverage is a huge improvement from the zero automated coverage we have today.

Multi-account scalability is really nice. Currently, only five child accounts are being used, but that number can be scaled at will to whatever number we please (within AWS Organization resource limits for child accounts). Five accounts is enough for us to run all e2e tests simultaneously without incident (though rate limit issues can still develop if all tests are run multiple times simultaneously). Hopefully this makes shepherding releases a bit less painful.

#### Description of how you validated changes

See https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/2764/workflows/bc163b82-bb01-404e-826d-902bee01f129

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by edwardfoyle** [comment]
Also, what's the story on resource cleanup with this change? Does the automated cleanup that runs in CCI still work? When we need to manually clean resources will we have to assume a role in each child account to clean up?

File: .circleci/local_publish_helpers.sh [hunk 1]
```diff
@@ -52,3 +52,48 @@ function setSudoNpmRegistryUrlToLocal {
   sudo npm set registry "$custom_registry_url"
   sudo yarn config set registry "$custom_registry_url"
 }
+
+function setAwsAccountCredentials {
```
[the remark this answers is not in the document]
Comment (johnpc) [reply .circleci/local_publish_helpers.sh:56]: Unfortunately, I couldn't get JS to behave how I wanted. The e2e test script runs `configure_tests.js && jest`. For this to work right, the configuration step needs to set the environment variable across the entire command, but JS doesn't work that way. I originally, I had gross js workarounds to make it work (see first draft windows pr) but it really wasn't worth the headache.

File: .circleci/local_publish_helpers.sh [hunk 2]
```diff
@@ -52,3 +52,48 @@ function setSudoNpmRegistryUrlToLocal {
...
+  if [[ "$OSTYPE" == "win32" ]]; then
+    # windows, do nothing
+    echo "Using parent account credentials for OSTYPE $OSTYPE"
+  elif [[ "$OSTYPE" == "cygwin" ]]; then
+    # windows, do nothing
+    echo "Using parent account credentials for OSTYPE $OSTYPE"
+  elif [[ "$OSTYPE" == "msys" ]]; then
+    # windows, do nothing
```
[the remark this answers is not in the document]
Comment (johnpc) [reply .circleci/local_publish_helpers.sh:64]: haha true - removed it!

**Comment by johnpc** [comment]
> Also, what's the story on resource cleanup with this change? Does the automated cleanup that runs in CCI still work? When we need to manually clean resources will we have to assume a role in each child account to clean up?

https://github.com/aws-amplify/amplify-cli/pull/8082 can be merged independently (I hoped it would be by now)

**Pull request merged**